File: leetcode/ValidPalindrome.java
/**
 * Valid Palindrome
 * https://leetcode.com/problems/valid-palindrome/
 */
class Solution {

    /**
     * @param String $s
     * @return Boolean
     */
    function isPalindrome($s) {
        $replaced = preg_replace('/[^a-z0-9]/','', strtolower($s));
        
        if (empty($replaced)  || strlen($replaced) == 1) {
            return true;
        }
        
        $front = 0;
        $end = strlen($replaced) - 1;
        
        while ($front < $end) {
            if ($replaced[$front] !== $replaced[$end]) {
                return false;
            }
            
            $front = $front+1;
            $end = $end-1;
        }
        
        return true;
   

File: leetcode/ValidPalindrome.js
/**
 * https://leetcode.com/problems/valid-palindrome/
 * 
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    // Remove non-alphanumeric characters and convert to lowercase
    let stripped = s.replace(/[^0-9a-z]/gi, '').toLowerCase();
    // Reverse the string and compare it to the original
    let reversed = stripped.split('').reverse().join('');
    return stripped === reversed;
};
